ruby/ql/lib/codeql/ruby/frameworks/ActionController.qll

```diff
@@ -92,12 +92,21 @@ class ActionControllerActionMethod extends Method, HTTP::Server::RequestHandler:
    * May return multiple results.
    */
   ActionDispatch::Route getARoute() {
-    result.getController() + "_controller" =
-      ActionDispatch::underscore(namespaceDeclaration(controllerClass)) and
-    isActionControllerMethod(this, result.getAction(), controllerClass)
+    exists(string name |
+      isRoute(result, name, controllerClass) and
+      isActionControllerMethod(this, name, controllerClass)
+    )
   }
 }
 
+private predicate isRoute(
```

Comment:
I suggest adding `pragma[nomagic]` to ensure that the compiler does not mess up our optimization (e.g. by inlining the predicate)

Reply:
Done